### ISSUE TITLE
reduce with an inappropriate function throws the wrong error

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#4189]
```
r.expr([1,2,3]).reduce(function(left) {
  return left
})
// Throws
RqlRuntimeError: Expected 1 argument but found 2 in:
// Instead of `RqlRuntimeError: Expected 2 arguments but found 1 in:`

r.expr([1,2,3]).reduce(function(left, right, extra) {
  return left
})
//Throws
RqlRuntimeError: Expected 3 arguments but found 2 in:
// Instead of `RqlRuntimeError: Expected 2 arguments but found 3 in:`
```

**Comment by neumino** [comment]
The same happens with `filter` (and I guess other methods?)

**Comment by mlucy** [comment]
What's the expected error?

**Comment by Tryneus** [comment]
@mlucy, the printf parameters are switched

**Comment by danielmewes** [comment]
The current error messages make sense to me. I don't think they are wrong.

As an improvement, we could check the arity of functions before we call them, and then print something like "Expected a 2-argument function but found a 3-argument function in:..."

**Comment by mlucy** [comment]
My interpretation of the current error message is "This function expected 3 arguments but was passed 2."  We should probably change the message to be clearer if it confused two people.

**Comment by mlucy** [comment]
This was fixed by @fantasticsid in PR #4644 (reviewed by me).
